lib/db/queries.ts

```diff
@@ -0,0 +1,159 @@
+import { sql, and, gte, eq, lte, not, isNull } from 'drizzle-orm';
+import { db } from './drizzle';
+import { books, authors, bookToAuthor } from './schema';
+import { SearchParams } from '@/lib/url-state';
+
+export const ITEMS_PER_PAGE = 28;
+export const EMPTY_IMAGE_URL =
+  'https://s.gr-assets.com/assets/nophoto/book/111x148-bcc042a9c91a29c1d680899eff700a03.png';
+
+const yearFilter = (yr?: string) => {
+  if (yr) {
+    const maxYear = Math.max(1950, Math.min(2023, Number(yr)));
+    return and(
+      gte(books.publication_year, 1950),
+      lte(books.publication_year, maxYear)
+    );
+  }
+  return and(
+    gte(books.publication_year, 1950),
+    lte(books.publication_year, 2023)
+  );
+};
+
+const ratingFilter = (rtg?: string) => {
+  if (rtg) {
+    const minRating = Number(rtg);
+    return sql`${books.average_rating} >= ${minRating}`;
+  }
+  return undefined;
+};
+
+const languageFilter = (lng?: string) => {
+  if (lng === 'en') {
+    return sql`${books.language_code} IN ('eng', 'en-US', 'en-GB')`;
+  }
+  return lng ? eq(books.language_code, lng) : undefined;
+};
+
+const pageFilter = (pgs?: string) => {
+  if (pgs) {
+    const maxPages = Math.min(3000, Number(pgs));
+    return lte(books.num_pages, maxPages);
+  }
+  return lte(books.num_pages, 3000);
+};
+
+const searchFilter = (q?: string) => {
+  if (q) {
+    const tsQuery = q.trim().split(/\s+/).join(' & ');
+    return sql`${books.title_tsv} @@ to_tsquery('english', ${tsQuery})`;
+  }
+  return undefined;
+};
+
+const imageFilter = () => {
+  return and(
+    not(isNull(books.image_url)),
+    sql`${books.image_url} != ${EMPTY_IMAGE_URL}`
+  );
+};
+
+const isbnFilter = (isbn?: string) => {
+  if (isbn) {
+    const isbnArray = isbn.split(',').map((id) => id.trim());
+    return sql`books.isbn IN (${sql.join(
```

Comment:
maybe we should use `inArray`?
please let me know if that didn't work for some reason